kit/ForKit.cpp

```diff
@@ -281,39 +281,56 @@ static void cleanupChildren()
-                                        WTERMSIG(status) == SIGBUS ||
-                                        WTERMSIG(status) == SIGABRT))
+            if (WIFSIGNALED(status))
             {
-                ++segFaultCount;
-
-                std::string noteCrashFile(it->second + "/tmp/kit-crashed");
-                int noteCrashFD = open(noteCrashFile.c_str(), O_CREAT | O_TRUNC | O_WRONLY, S_IRUSR | S_IWUSR);
-                if (noteCrashFD < 0)
-                    LOG_ERR("Couldn't create file: " << noteCrashFile << " due to error: " << strerror(errno));
+                if (WTERMSIG(status) == SIGSEGV || WTERMSIG(status) == SIGBUS ||
+                    WTERMSIG(status) == SIGABRT)
+                {
+                    ++segFaultCount;
+                    LOG_INF("Child " << exitedChildPid << " has crashed, will remove its jail ["
+                                     << it->second << "].");
+
+                    std::string noteCrashFile(it->second + "/tmp/kit-crashed");
+                    int noteCrashFD = open(noteCrashFile.c_str(), O_CREAT | O_TRUNC | O_WRONLY,
+                                           S_IRUSR | S_IWUSR);
+                    if (noteCrashFD < 0)
+                        LOG_ERR("Couldn't create file: " << noteCrashFile
+                                                         << " due to error: " << strerror(errno));
+                    else
+                        close(noteCrashFD);
+                }
+                else if (WTERMSIG(status) == SIGKILL || WTERMSIG(status) == SIGTERM)
+                {
+                    ++killedCount;
+                    LOG_INF("Child " << exitedChildPid << " was killed, will remove its jail ["
+                                     << it->second << "].");
+                }
                 else
-                    close(noteCrashFD);
-            }
+                {
+                    LOG_ERR("Child " << exitedChildPid << " has exited, with status" << status);
+                }
 
-            LOG_INF("Child " << exitedChildPid << " has exited, will remove its jail [" << it->second << "].");
-            cleanupJailPaths.emplace_back(it->second);
-            childJails.erase(it);
-            if (childJails.empty() && !SigUtil::getTerminationFlag())
-            {
-                // We ran out of kits and we aren't terminating.
-                LOG_WRN("No live Kits exist, and we are not terminating yet.");
+                LOG_INF("Child " << exitedChildPid << " has exited, will remove its jail ["
+                                 << it->second << "].");
+                cleanupJailPaths.emplace_back(it->second);
+                childJails.erase(it);
+                if (childJails.empty() && !SigUtil::getTerminationFlag())
+                {
+                    // We ran out of kits and we aren't terminating.
+                    LOG_WRN("No live Kits exist, and we are not terminating yet.");
+                }
             }
```

Comment:
By splitting the if (WIFSIGNALED() check - surely we changed the behavior here. We no longer warn and clean up in the !WIFSiGNALED() case ?

Is that expected, deliberate etc. ?

Reply:
It was missing.